### PR TITLE
7pm-2-ShowSearchQuota

### PR DESCRIPTION
**Pull request by bowenduanmu**
Show user search quota before the first search in Google Search page of the app.

### CODE REVIEW

**Comment by pconrad** [comment]
Note: there is a missed line of coverage on the codecov patch... it says that line 62 is not covered by tests.

THAT IS OK.  We are going to let that slide.   Getting to the tests passing at all was hard fought, and it's getting late in the quarter.

File: javascript/src/main/pages/Search/Search.js [hunk 1]
```diff
@@ -46,13 +47,24 @@ const Search = () => {
...
     const [results, setResults] = useState(emptyResults);
     const [quota, setQuota] = useState(0);
 
+    useEffect(() => {
+        async function getQuota() {
+            const quotaInfo = await fetchQuota();
+            console.log("quotaInfo=",quotaInfo)
+            if(quotaInfo && quotaInfo.quota)
+                setQuota(quotaInfo.quota);
+        };
```
Comment (pconrad) [inline javascript/src/main/pages/Search/Search.js:63]: This line not being covered by tests?  We are going to let that slide!!!  Sincerely, Prof. Conrad. Kthxbye.

**Review by pconrad — APPROVED** [review]
LGTM

File: javascript/src/test/pages/Search/Search.test.js [hunk 2]
```diff
@@ -27,25 +25,40 @@ describe("Search tests", () => {
...
         await waitFor(() => expect(fetchWithToken).toHaveBeenCalled());
     });
 
 
     test("renders when submit button is pressed and results have contents", async () => {
-        fetchWithToken.mockResolvedValue(fakeResults);
+        
+        fetchWithToken.mockImplementation(
+            (url) => {
+                console.log(`url=${url}`)
```
Comment (pconrad) [inline javascript/src/test/pages/Search/Search.test.js:49]: Please remove this console.log

File: javascript/src/test/pages/Search/Search.test.js [hunk 3]
```diff
@@ -27,25 +25,40 @@ describe("Search tests", () => {
     });
 
     test("renders when submit button is pressed and results are empty", async () => {
-        fetchWithToken.mockResolvedValue({items:[]});
+
+        fetchWithToken.mockImplementation(
+            (url) => {
+                console.log(`url=${url}`)
```
Comment (pconrad) [inline javascript/src/test/pages/Search/Search.test.js:31]: Please remove this console.log

File: javascript/src/main/pages/Search/Search.js [hunk 4]
```diff
@@ -46,13 +47,24 @@ const Search = () => {
...
 
     
     const [query, setQuery] = useState(emptyQuery);
     const [results, setResults] = useState(emptyResults);
     const [quota, setQuota] = useState(0);
 
+    useEffect(() => {
+        async function getQuota() {
+            const quotaInfo = await fetchQuota();
+            console.log("quotaInfo=",quotaInfo)
```
Comment (pconrad) [inline javascript/src/main/pages/Search/Search.js:60]: Please remove this console.log

File: javascript/src/main/pages/Search/Search.js [hunk 5]
```diff
@@ -46,13 +47,24 @@ const Search = () => {
             console.log(`err=${err}`);
```
Comment (pconrad) [inline javascript/src/main/pages/Search/Search.js:47]: This console.log is ok and can stay, since it is covering an error condition.

**Review by pconrad — CHANGES_REQUESTED** [review]
There are a couple of console.log statements that we should remove.

After that, it looks good to me!

**Review by ujcookcao — APPROVED** [review]
The code successfully shows the quota in the google search function and passes all the tests.

**Review by pconrad — APPROVED** [review]
LGTM

**Review by ujcookcao — APPROVED** [review]
The code successfully shows the quota in the google search function and passes all the tests.

**Comment by andrewhlu** [comment]
@pconrad I see that you previously said you were okay with overriding test coverage drops. If that's the case, feel free to merge.

**Pull request merged**